Add `sentry-jdbc` guide. (#4338)
getsentry/sentry-docs: src/platforms/java/performance/instrumentation/jdbc.mdx

---
title: JDBC Instrumentation
sidebar_order: 20
description: "Learn how to capture the performance of database queries executed with JDBC."
---

<Note>

Capturing transactions requires that you first <PlatformLink to="/performance/">set up performance monitoring</PlatformLink> if you haven't already.

</Note>

> Sentry JDBC integration provides the `SentryJdbcEventListener` for [P6Spy](https://github.com/p6spy/p6spy/) - database activity interceptor - which creates a span for each JDBC statement executed over proxied instance of `javax.sql.DataSource`.

Reviewer: ```suggestion
Sentry JDBC integration provides the `SentryJdbcEventListener` for [P6Spy](https://github.com/p6spy/p6spy/)  database activity interceptor, which creates a span for each JDBC statement executed over a proxied instance of `javax.sql.DataSource`.
```

### Install

```xml {tabTitle:Maven}
<dependency>
    <groupId>io.sentry</groupId>
    <artifactId>sentry-jdbc</artifactId>
    <version>{{ packages.version('sentry.java.openfeign', '5.3.0') }}</version>
</dependency>
```

```groovy {tabTitle:Gradle}
implementation 'io.sentry:sentry-jdbc:{{ packages.version('sentry.java.jdbc', '5.3.0') }}'
```

```scala {tabTitle: SBT}
libraryDependencies += "io.sentry" % "sentry-jdbc" % "{{ packages.version('sentry.java.jdbc', '5.3.0') }}"
```

> For other dependency managers see the [central Maven repository](https://search.maven.org/artifact/io.sentry/sentry-jdbc).

Reviewer: ```suggestion
For other dependency managers, check out the [central Maven repository](https://search.maven.org/artifact/io.sentry/sentry-jdbc).
```

> Configure `DataSource` to use `com.p6spy.engine.spy.P6SpyDriver` as a JDBC driver. For Spring Boot applications:

Reviewer 2: ```suggestion
Configure the `DataSource` to use `com.p6spy.engine.spy.P6SpyDriver` as a JDBC driver. For Spring Boot applications:
```
```suggestion
Configure `DataSource` to use `com.p6spy.engine.spy.P6SpyDriver` as a JDBC driver. For Spring Boot applications:
```

```properies {tabTitle: application.properties}
spring.datasource.driver-class-name=com.p6spy.engine.spy.P6SpyDriver
```

```yaml {tabTitle: application.yml}
spring.datasource.driver-class-name: com.p6spy.engine.spy.P6SpyDriver
```

> Add `p6spy` prefix to database connection url. For Spring Boot applications:

Reviewer: ```suggestion
Add the `p6spy` prefix to the database connection URL. For Spring Boot applications:
```

```properies {tabTitle: application.properties}

> spring.datasource.url: jdbc:p6spy:...

Reviewer 2: What about an example instead of `...`?

Author: 👍

```

```yaml {tabTitle: application.yml}
spring.datasource.url: jdbc:p6spy:...
```

> P6Spy uses a Java [Service Loader](https://docs.oracle.com/javase/tutorial/sound/SPI-intro.html) mechanism to register `JdbcEventListener`s. For more configuration options, check [P6Spy Reference Guide](https://p6spy.readthedocs.io/en/latest/).

Reviewer: ```suggestion
P6Spy uses a Java [Service Loader](https://docs.oracle.com/javase/tutorial/sound/SPI-intro.html) mechanism to register `JdbcEventListener`s. For more configuration options, check the [P6Spy Reference Guide](https://p6spy.readthedocs.io/en/latest/).
```
